packages/common/src/intl/intlHelpers.ts

```diff
@@ -24,6 +26,81 @@ export const useTranslation = (ns?: Namespace): TypedTFunction<LocaleKey> => {
   return (key, options) => (key ? t(key, options) : '');
 };
 
+// This custom formatter will truncate cents to two digits when there are none.
+// However, still allows a large precision.
+// Without this custom formatter, a value of 12 would be formatted as
+// 12.00000 if the precision was 5, for example. With this change, the format
+// would be 12.00. With a precision of 2, then currency values would be
+// truncated to be a maximum of two decimal places.
+const formatter: currency.Format = (currency?, options?): string => {
+  const { decimal = '.' } = options ?? {};
+  if (!currency) return '';
+  if (!currency.cents()) {
+    return `${currency.value}${decimal}${'0'.repeat(2)}`;
+  }
+  return String(parseFloat(String(currency.value)));
+};
+
+const currencyOptions = {
+  en: {
+    symbol: '$',
+    separator: ',',
+    decimal: '.',
+    precision: 10,
+    pattern: '!#',
+    negativePattern: '-!#',
+    format: formatter,
+  },
+  fr: {
+    symbol: 'XOF',
+    separator: '.',
+    decimal: ',',
+    precision: 10,
+    pattern: '# !',
+    negativePattern: '-!#',
+    format: formatter,
+  },
+  ar: {
+    symbol: 'ر.ق.',
+    separator: ',',
+    decimal: '.',
+    precision: 10,
+    pattern: '!#',
+    negativePattern: '-!#',
+    format: formatter,
+  },
```

Comment:
seems that if we were to use `intl` we would need *something* like this anyway, but would also maybe need to do more work. I.e. I can't get `Intl` to format with a `XOF` suffix, so might need a customer`format` method to handle that.

I don't know what the future holds for currencies, but I imagine either having these functions also convert to the actual currency from some base, or only displaying a single currency per deployment, in which case this config could come from the server on login

Reply:
when using `intl` this wouldn't be required for most locales - only the edge cases. 
It is a lot more painful to customise when required though, and so this is better long-term I think.

My main concern is that the options are country-specific and not language-specific, as they are here. We do need a way to have the a language translation set to be used, with a different currency format option.